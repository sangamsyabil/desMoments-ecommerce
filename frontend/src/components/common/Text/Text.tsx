import { makeStyles, Typography } from '@material-ui/core';
import clsx from 'clsx';
import React from 'react';
import { CustomTheme } from '../../../theme/muiTheme';

export interface TextProps {
  externalStyle?: string;
  variant?:
    | 'button'
    | 'caption'
    | 'h1'
    | 'h2'
    | 'h3'
    | 'h4'
    | 'h5'
    | 'h6'
    | 'inherit'
    | 'subtitle1'
    | 'subtitle2'
    | 'body1'
    | 'body2'
    | 'overline'
    | 'srOnly'
    | undefined;
  color?: 'inherit' | 'initial' | 'primary' | 'secondary' | 'textPrimary' | 'textSecondary' | 'error' | undefined;
  fontWeight?: 'bold' | 'normal';
  textTransform?: 'uppercase' | 'lowercase' | 'capitalize';
}

const useStyles = makeStyles((theme: CustomTheme) => ({
  text: {
    fontWeight: ({ fontWeight }: TextProps): any => (fontWeight ? fontWeight : 'normal'),
    textTransform: ({ textTransform }: TextProps): any => (textTransform ? textTransform : 'lowercase'),
  },
}));

const Text: React.FC<TextProps> = (props) => {
  const classes = useStyles(props);
  const { externalStyle, children, variant, color } = props;
  return (
    <Typography variant={variant} color={color} className={clsx([classes.text, externalStyle])}>
      {children}
    </Typography>
  );
};

export { Text };
