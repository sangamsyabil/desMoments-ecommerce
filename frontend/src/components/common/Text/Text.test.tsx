import { render } from '@testing-library/react';
import React from 'react';
import { Text, TextProps } from './Text';

const mockData: TextProps = {
  variant: 'button',
  color: 'initial',
  textTransform: 'uppercase',
  fontWeight: 'bold',
};

test('Text renders correctly', () => {
  const { asFragment } = render(<Text {...mockData}>Mock button</Text>);
  expect(asFragment()).toMatchSnapshot();
});
