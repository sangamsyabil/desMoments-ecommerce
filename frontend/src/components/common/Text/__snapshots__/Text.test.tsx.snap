// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<DocumentFragment>
  <span
    class="MuiTypography-root makeStyles-text-1 makeStyles-text-2 MuiTypography-button"
  >
    Mock button
  </span>
</DocumentFragment>
`;
